packages/x-adapter/src/mappers/schema-mapper.factory.ts

```diff
@@ -95,16 +100,35 @@ function applySubSchemaTransformer<Source, Target>(
   schema: Schema<Source, Target>
 ): Target | Target[] | undefined {
   const subSource = extractValue(source, $path);
-  const context = deepMerge(rawContext, $context);
-  const subSchema = $subSchema === '$self' ? schema : $subSchema;
 
-  if (subSource) {
-    return isArray(subSource)
-      ? subSource.map(item => mapSchema(item, subSchema, context) as Target)
-      : mapSchema<typeof subSource, Target>(
-          subSource,
-          subSchema as Schema<typeof subSource, Target>,
-          context
-        );
+  if (!subSource) {
+    return;
+  }
+
+  const extendedContext: Dictionary = {};
+  if ($context) {
+    Object.entries($context).forEach(([key, value]) => {
+      if (['requestParameters', 'endpoint', 'to'].includes(key)) {
+        return;
+      }
+      extendedContext[key] = extractValue(source, value as ExtractPath<typeof source>);
+    });
+  }
```

Comment:
You are probably hating me right now, but... can we refactor this to the following?

```suggestion
  const extendedContext = Object.entries($context ?? []).reduce(
    (extendedContext, [key, value]) => ({
      ...extendedContext,
      // eslint-disable-next-line @typescript-eslint/no-extra-parens
      ...(['requestParameters', 'endpoint', 'to'].includes(key)
        ? { key: extractValue(source, value as ExtractPath<typeof source>) }
        : {})
    }),
    {} as Dictionary
  );
```